stdlib/public/core/CString.swift

```diff
@@ -198,8 +230,21 @@ extension String {
     self = str
   }
 
+  /// Creates a new string by copying and validating the null-terminated UTF-8
+  /// data referenced by the given array.
+  ///
+  /// This initializer does not try to repair ill-formed UTF-8 code unit
+  /// sequences. If any are found, the result of the initializer is `nil`.
+  ///
+  /// Note: This initializer is deprecated. Use the initializer
+  ///       `String.init?(validating: array, as: UTF8.self)` instead,
+  ///       remembering that "\0" is a valid character in Swift.
+  ///
+  /// - Parameter nullTerminatedUTF8:
+  ///     An array containing a null-terminated sequence of UTF-8 code units.
   @inlinable
   @_alwaysEmitIntoClient
+  @available(swift, deprecated: 6, message: "Use String(validating: array, as: UTF8.self) instead")
```

Comment:
As with the other initializer, the null character poses a problem. Since this initializer is basically _never_ desireable, is it possible to mark it `unavailable` rather than deprecated, or will that prevent it from participating in overload resolution (despite being an exact match)?

Reply:
Assuming the overload resolution still picks it, anyway.

Reply:
Your intuition was correct: overload resolution skips over unavailable or obsoleted overloads. Oh well.